### ISSUE TITLE
How to align the messageInputBar with the chat view?

### ISSUE THREAD

**Issue by aadjemonkeyrock** [MessageKit/MessageKit#929]
I'm moving a iPhone/iPad project from JSQMessagesViewController to MessageKit. This is going very well, except for a small layout issue on my iPad app. I've all the functionality, layout etc. working fine. 

I'm using MessageKit 2.0.0, on various iOS versions in simulators and devices. On the iPhone everything is fine, but on the iPad I've a layout issue due to the screen design. The iPad as a left site menu (kind of a left site tabbar). Clicking on the left side menu brings forward a new view controller. 

With JSQMessageViewController this looks like this
![img_0184](https://user-images.githubusercontent.com/956176/48060279-dc8a8680-e1bb-11e8-9f1b-b585927ec657.PNG)

and with the keyboard expanded
![img_0185](https://user-images.githubusercontent.com/956176/48060301-ef9d5680-e1bb-11e8-9ca5-9d2e473061a1.PNG)

As you can see the MessageInputBar stays within the boundaries of the chat view controller. 

With MessageKit I can't replicate this behaviour, or at least I don't know how to do this. The result is that the MessageInputBar is always the full width. 

<img width="611" alt="ipad_air_2_-_12_1_and_slack_-_bitcare___humanwave" src="https://user-images.githubusercontent.com/956176/48060411-2e331100-e1bc-11e8-81a2-238dc57ae874.png">

Is there a way to fix this? 

Thx
-Aad

**Comment by nathantannar4** [comment]
As of now, this is currently not possible. I can look into possible resolutions. As a temp fix, you could move that refresh button up a bit so its not hidden.

**Comment by nathantannar4** [comment]
@aadjemonkeyrock I implemented a possible solution on my InputBarAccessoryView, the project MessageInputBar has much of its implementation forked from. Before I make any merges to MessageInputBar, can you try testing it for me?

https://github.com/nathantannar4/InputBarAccessoryView/pull/43

Add this to your Podfile
pod 'InputBarAccessoryView', :git => 'https://github.com/nathantannar4/InputBarAccessoryView.git', :branch => 'frameInsets'

Then run pod update and `import InputBarAccessoryView` to where you have your `MessagesViewController`

Override the `inputAccessoryView` of your controller to be an instance of `InputBarAccessoryView`

Then set the frame insets of the `InputBarAccessoryView`

```
class MyVC: MessagesViewController {

let inputBar = InputBarAccessoryView()

override var inputAccessoryView: UIView? { return inputBar }

override func viewDidLoad() {
super.viewDidLoad()
...
let leftInset: CGFloat = 30 // whichever the value of the inset is for the sidebar you have
inputBar.frameInsets = HorizontalEdgeInsets(left: leftInset, right: 0)
}

}
```

**Comment by aadjemonkeyrock** [comment]
@nathantannar4 Thank you for the effort. Excuus it took a bit of time b4 I could look into it. 

I do have the inset working, that is looking good. However I did have a conflict with 'HorizontalEdgeInsets' being defined twice, both in the InputBarAccessoryView and MessageKit. To make it work I had to implement an intializer in the InputBarAccessoryView version.

It's looking good, thx for the effort
